src/main/java/cloud/eppo/EppoClient.java

```diff
@@ -0,0 +1,477 @@
+import cloud.eppo.ufc.dto.EppoValue;
+import cloud.eppo.ufc.dto.FlagConfig;
+import cloud.eppo.ufc.dto.SubjectAttributes;
+import cloud.eppo.ufc.dto.VariationType;
+import java.util.HashMap;
+import java.util.Map;
+
+import cloud.eppo.ufc.dto.adapters.EppoModule;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EppoClient {
+  private static final Logger log = LoggerFactory.getLogger(EppoClient.class);
+  private final ObjectMapper mapper = new ObjectMapper().registerModule(EppoModule.eppoModule()); // TODO: is this the best place for this?
+  private static final String DEFAULT_HOST = "https://fscdn.eppo.cloud";
+  private static final boolean DEFAULT_IS_GRACEFUL_MODE = true;
+
+  private final ConfigurationRequestor requestor;
+  private final AssignmentLogger assignmentLogger;
+  private final boolean isGracefulMode;
+  private final String sdkName;
+  private final String sdkVersion;
+  private final boolean isConfigObfuscated;
+
+  private static EppoClient instance;
+
+  // Fields useful for testing in situations where we want to mock the http client or configuration
+  // store (accessed via reflection)
+  /**
+   * @noinspection FieldMayBeFinal
+   */
+  private static EppoHttpClient httpClientOverride = null;
+
+  /**
+   * @noinspection FieldMayBeFinal
+   */
+  private static ConfigurationStore configurationStoreOverride = null;
+
+  private EppoClient(
+      String apiKey,
+      String sdkName,
+      String sdkVersion,
+      String host,
+      AssignmentLogger assignmentLogger,
+      boolean isGracefulMode
+  ) {
+    ConfigurationStore configStore =
+        configurationStoreOverride == null
+            ? new ConfigurationStore()
+            : configurationStoreOverride;
+
+    EppoHttpClient httpClient = buildHttpClient(host, apiKey, sdkName, sdkVersion);
+    requestor = new ConfigurationRequestor(configStore, httpClient);
+    this.assignmentLogger = assignmentLogger;
+    this.isGracefulMode = isGracefulMode;
+    // Save SDK name and version to include in logger metadata
+    this.sdkName = sdkName;
+    this.sdkVersion = sdkVersion;
+    // For now, the configuration is only obfuscated for Android clients
+    this.isConfigObfuscated = sdkName.toLowerCase().contains("android");
+    // TODO: caching initialization (such as setting an API-key-specific prefix
+  }
+
+  private EppoHttpClient buildHttpClient(String host, String apiKey, String sdkName, String sdkVersion) {
+    EppoHttpClient httpClient;
+    if (httpClientOverride != null) {
+      // Test/Debug - Client is mocked entirely
+      httpClient = httpClientOverride;
+    } else {
+      // Normal operation
+      httpClient = new EppoHttpClient(host, apiKey, sdkName, sdkVersion);
+    }
+    return httpClient;
+  }
+
+  public static EppoClient init(
```

Comment:
My guess is this method will probably be phased out for just the constructor and each upstream SDK will have it's own `init()` and/or perhaps they will subclass this client.